linttest/linttest.go

```diff
@@ -95,27 +96,41 @@ func checkFiles(t *testing.T, c *lintpack.Checker, ctx *lintpack.Context, pkg *p
 	for _, f := range pkg.Syntax {
 		filename := getFilename(ctx.FileSet, f)
 		testFilename := filepath.Join("testdata", c.Info.Name, filename)
-		ws := newWarnings(t, testFilename)
+
+		var ws warnings
+		func() {
+			rc, err := os.Open(testFilename)
+			if err != nil {
+				t.Fatalf("read file %q: %w", testFilename, err)
+			}
+			defer rc.Close()
+
+			ws, err = newWarnings(rc)
+			if err != nil {
+				t.Fatal(err)
+			}
```

Comment:
I've used `readfile` before to avoid messing with file closing.
If you do the same here and create a reader out of `[]byte`, closure can be removed and no defers are needed.

Reply:
Yes, but I was hoping to avoid copying all the bytes an extra time just for the sake of that.

If I just factor out the loop instead (change `checkFiles` to `checkFile`), the closure's not needed and the whole thing gets less indented. I'll push a commit with that and you can see what you think.